cluster-autoscaler/core/scaledown/deletionbatcher/node_deletion_batcher.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deletionbatcher
+
+import (
+	apiv1 "k8s.io/api/core/v1"
+)
+
+// NodeBatcherDeleter is gathering nodes and deleting them in bathch.
+type NodeBatcherDeleter interface {
+	// AddToBucket adds node to delete candidates.
+	AddToBucket(node *apiv1.Node) (string, error)
```

Comment:
I see your point. 
In case we delete nodes in batch we don't delete individual nodes, we delete the whole batch. I initially thought that NodeBatcherDeleter is an interpretation of Bucket, where we can add nodes and call cleanup, but the logic what to add and when is on actuator side. 
However the idea to move the deletion logic to Batcher is good. I wouldn't name function as ScheduleNodeDeletion because we already have scheduleDeletion() method in actuator. I'd name it just AddNode().

Reply:
Yeah good idea, the type name already contains "delete" and "batch", so an `AddNode` method SGTM.